corgi/core/migrations/0081_fix_maven_namespaces.py

```diff
@@ -0,0 +1,35 @@
+from django.db import migrations
+from django.db.models import Value, functions
+
+
+def fix_maven_namespaces(apps, schema_editor):
+    """Set namespace to REDHAT for Maven components with "redhat" in their version strings"""
+    Component = apps.get_model("core", "Component")
+    ComponentNode = apps.get_model("core", "ComponentNode")
+
+    # Values in current data are always -redhat or .redhat
+    maven_components = Component.objects.filter(
+        type="MAVEN", namespace="UPSTREAM", version__contains="redhat"
+    )
+    upstream_maven = Value("pkg:maven/")
+    redhat_maven = Value("pkg:maven/redhat/")
```

Comment:
Instead of modifying the namespace with a `redhat/` suffix let's use the repository_url qualifier ie:

`pkg:maven/org.apache.xmlgraphics/batik-anim@1.9.1?repository_url=repo.spring.io%2Frelease`

From https://github.com/package-url/purl-spec/blob/master/README.rst#some-purl-examples

In our case the repository_url for Red Hat Maven repository is:

`maven.repository.redhat.com/ga/`

From https://access.redhat.com/maven-repository

Reply:
Using the GA repository seems like the only reasonable think to do, since we won't be able to tell which repository a component is destined for at build time. I'm confident most of the time the components will be destined for the GA repository though.